ui/storybook/README.md

```diff
@@ -81,12 +79,55 @@ Below are some notable addons this project currently uses.
 - [Docs](https://storybook.js.org/addons/@storybook/addon-docs/) - Document component usage and properties in Markdown.
 - [Links](https://storybook.js.org/addons/@storybook/addon-links/) - Use to create links that navigate between stories.
 - [Measure](https://storybook.js.org/addons/@storybook/addon-measure/) - Helpful for inspecting layouts by visualizing the box model.
+- [Mock Service Worker](https://storybook.js.org/addons/msw-storybook-addon) - Allows mocking of api calls using [Mock Service Worker](https://mswjs.io/).
 - [Outline](https://storybook.js.org/addons/@storybook/addon-outline/) - Outline all elements with CSS to help with layout placement and alignment.
 - [Storysource](https://storybook.js.org/addons/@storybook/addon-storysource/) - Used to show stories source in the addon panel
 - [Viewport](https://storybook.js.org/addons/@storybook/addon-viewport/) - Allows stories to be displayed in different sizes and layouts.
 
 We primarily use first party addons maintained by Storybook to avoid pain with upgrades and interoperability with the relatively complex Storybook ecosystem.
 
+## Defining and using decorators
+
+[Decorators](https://storybook.js.org/docs/react/writing-stories/decorators) provide a way to wrap a story in extra "rendering" functionality.
+
+### Global decorators
+
+Global decorators are specified in the [Storybook configuration](#configuring-storybook) and are applied to all stories. These decorators live in the `storybook` package in `src/config/decorators` and are primarily related to global storybook behavior (e.g. theming, time zones).
+
+### Package-specific decorators
+
+Perses relies heavily on React context, which can be tedious to set up over and over again in every story. We simplify this by creating decorators to cover common cases related to React context. These decorators should:
```

Comment:
We’ve talked about wanting to split up and move some of what’s in our PersesProviders.tsx and PersesDashboardProviders.tsx internally into OSS. We’d have to split out the app-level providers and plugin registry, datasource provider, but would having something like that also be reusable for Storybook? It seems like every repo I embed Perses in needs an easy way to encapsulate the common ones like TimeRangeProvider, TemplateVariableProvider, etc.

Reply:
Well I guess TemplateVariableProvider is a bad example due to circular dep since it has to live in dashboards but the ones that already live in plugin-system might make sense to have exported out nicely in a PluginSystemProviders.tsx for embed use cases